Missing C++ dependency in README & System tray icon installed as regular file instead of symlink

I found two issues during installation:

Both issues were tested on Kubuntu 24.04/26.04

### 1. Missing C++ Compiler Dependency
* **Issue:** The README does not mention the C++ compiler (`g++`) in the installation instructions. It should be added so users know it's required for building.

error log
```
$ cmake -B build -DCMAKE_INSTALL_PREFIX=/usr -DCMAKE_BUILD_TYPE=Release
-- The C compiler identification is GNU 15.2.0
-- The CXX compiler identification is unknown
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
CMake Error at CMakeLists.txt:2 (project):
  No CMAKE_CXX_COMPILER could be found.

  Tell CMake where to find the compiler by setting either the environment
  variable "CXX" or the CMake cache entry CMAKE_CXX_COMPILER to the full path
  to the compiler, or to the compiler name if it is in the PATH.
```

### 2. System Tray Icon Installed as Regular File (Not Symlink)
* **Issue:** In the repository, `fcitx_mcbopomofo.png` is a symbolic link. However, during installation, CMake's `install(DIRECTORY)` dereferences this symlink and installs it as a regular file (a duplicate copy of `org.fcitx.Fcitx5.fcitx_mcbopomofo.png`).
* **Impact:** This breaks the system tray icon in some desktop environments.
* **Expected Behavior:** `fcitx_mcbopomofo.png` should be correctly installed as a symbolic link pointing to `org.fcitx.Fcitx5.fcitx_mcbopomofo.png` at the destination.

--- a/README.md
+++ b/README.md
@@ -30,7 +30,7 @@
 
 ```bash
 sudo apt install \
-    pkg-config fcitx5 libfcitx5core-dev libfcitx5config-dev libfcitx5utils-dev fcitx5-modules-dev \
+    g++ pkg-config fcitx5 libfcitx5core-dev libfcitx5config-dev libfcitx5utils-dev fcitx5-modules-dev \
     cmake extra-cmake-modules gettext libfmt-dev libicu-dev libjson-c-dev
 ```
 

